core/loader_shared.c

```diff
@@ -410,6 +413,10 @@ privload_insert(privmod_t *after, app_pc base, size_t size, const char *name,
      */
     if (IF_UNIX_ELSE(mod->name == NULL, false)) {
         mod->name = double_strrchr(mod->path, DIRSEP, ALT_DIRSEP);
+        /* XXX: mod->name contains the leading '/'. We probably don't want that,
+         * but it doesn't seem to break anything, leaving it for now. Other call
+         * sites of double_strrchr() do increment the '/' off the library name.
```

Comment:
> * Other call
> * sites of double_strrchr() do increment the '/' off the library name.

Oh, I think I misunderstood your earlier comment: I thought you were saying that all users of mod->name incremented past the /.  Instead you're saying that some other (unrelated?) code using strrchr is skipping slashes -- not sure that's relevant here so you must be talking about other loader code using strrchr to find mod->name??

Reply:
> not sure that's relevant here so you must be talking about other loader code using strrchr to find mod->name??

I'll remove the addition again.